server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -286,7 +291,12 @@ public void onFailure(Exception e) {
 
                     // Start the segments files upload
                     uploadNewSegments(localSegmentsPostRefresh, localSegmentsSizeMap, segmentUploadsCompletedListener);
-                    latch.await();
+                    if (latch.await(
+                        remoteStoreSettings.getClusterRemoteSegmentTransferTimeout().millis(),
+                        TimeUnit.MILLISECONDS
+                    ) == false) {
+                        throw new InterruptedException("Timeout while waiting for remote segment transfer to complete");
```

Comment:
1. Just a thought if InterruptedException is the right exception here? As we are not actually interrupting the thread here, it's getting timed-out. 
2. Should we handle `InterruptedException` before we hit this timeout limit?

Reply:
1. Agree, i don't think we have a generic exception for segment upload failures. creating one!
2. `InterruptedException` would be handled by existing catch blocks and as of now i dont think we need to do any special handling in case of `InterruptedException`, a retry would be scheduled automatically.